http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java

```diff
@@ -1017,9 +1023,26 @@ static void recordRequestComplete(RequestStats requestStats, Request request, lo
 
         requestStats.recordResponseReceived(request.getMethod(),
                 response.getStatusCode(),
-                response.getBytesRead(),
+                requestBytes,
                 response.getBytesRead(),
                 requestProcessingTime,
                 responseProcessingTime);
     }
+
+    private static class RequestSizeListener
+            implements org.eclipse.jetty.client.api.Request.ContentListener
+    {
+        private long bytes;
```

Comment:
Is this the correct thread safety for a content listener callback?

Reply:
This matches the other listeners, so it should be ok.